src/test/java/org/opentripplanner/routing/trippattern/DeduplicatorTest.java

```diff
@@ -227,4 +234,66 @@ public void testToString() {
       subject.toString()
     );
   }
+
+  static List<Supplier<I18NString>> testCases() {
+    return List.of(
+      DeduplicatorTest::nonLocalizedString,
+      DeduplicatorTest::localizedString,
+      DeduplicatorTest::translatedString
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("testCases")
+  void deduplicateI18nString(Supplier<I18NString> makeString) {
+    var cache = new Deduplicator();
+
+    var s1 = makeString.get();
+    var s2 = makeString.get();
+    assertEquals(s1, s2);
+    assertNotSame(s1, s2);
+
+    var d1 = cache.deduplicateObject(I18NString.class, s1);
+    assertSame(s1, d1);
+
+    var d2 = cache.deduplicateObject(I18NString.class, s2);
+    assertEquals(s1, d2);
+
+    assertSame(s1, d2);
+    assertEquals(d1, d2);
+    assertSame(d1, d2);
+
+    var s3 = I18NString.of("Rue de Helmut Kohl");
```

Comment:
This has no functional impact at all, but since the spelling of "Gatan" was patched I thought I'd mention it: French streets named after people typically don't use "de". Example of an actual rue Helmut Kohl: https://www.openstreetmap.org/way/198061143
"Rue" is also not capitalized in most usage, but for some reason it is capitalized on maps (and thus in OpenStreetMap).